transport/socks5/stream_dialer_test.go

```diff
@@ -164,3 +167,84 @@ func testExchange(tb testing.TB, listener *net.TCPListener, destAddr string, req
 
 	running.Wait()
 }
+
+func TestConnectWithoutAuth(t *testing.T) {
+	// Create a SOCKS5 server
+	server := socks5.NewServer()
+	host := "127.0.0.1"
+
+	// Create SOCKS5 proxy on localhost with a random port
+	listener, err := net.Listen("tcp", net.JoinHostPort(host, "0"))
```

Comment:
This is more readable:
```suggestion
	listener, err := net.Listen("tcp", "127.0.0.1:0")
```